hudi-common/src/main/java/org/apache/hudi/common/util/CleanerUtils.java

```diff
@@ -139,7 +139,10 @@ public static Option<HoodieInstant> getEarliestCommitToRetain(
               if (nthInstant.compareTo(earliestPendingCommits.get()) <= 0) {
                 return Option.of(nthInstant);
               } else {
-                return completedCommitsTimeline.findInstantsBefore(earliestPendingCommits.get().getTimestamp()).lastInstant();
+                return earliestPendingCommits.map(earliestCommit ->
```

Comment:
We could rename earliestPendingCommits -> earliestPendingCommits since it is singular